corehq/apps/app_manager/views/apps.py

```diff
@@ -346,28 +346,43 @@ def copy_app(request, domain):
         else:
             app_id_or_source = app_id
 
-        def _inner(request, domain, data):
-            clear_app_cache(request, domain)
+        def _inner(request, link_domain, data, master_domain=domain):
+            clear_app_cache(request, link_domain)
             if data['toggles']:
                 for slug in data['toggles'].split(","):
-                    set_toggle(slug, domain, True, namespace=toggles.NAMESPACE_DOMAIN)
-            extra_properties = {'name': data['name']}
+                    set_toggle(slug, link_domain, True, namespace=toggles.NAMESPACE_DOMAIN)
             linked = data.get('linked')
             if linked:
-                extra_properties['master'] = app_id
-                extra_properties['doc_type'] = 'LinkedApplication'
-                if domain not in app.linked_whitelist:
-                    app.linked_whitelist.append(domain)
+                master_version = get_latest_released_app_version(app.domain, app_id)
+                if not master_version:
+                    messages.error(request, _("Creating linked app failed."
+                                              " Unable to get latest released version of your app."
+                                              " Make sure you have at least one released build."))
+                    return HttpResponseRedirect(reverse_util('app_settings', params={}, args=[domain, app_id]))
+
+                if link_domain not in app.linked_whitelist:
+                    app.linked_whitelist.append(link_domain)
                     app.save()
-            app_copy = import_app_util(app_id_or_source, domain, extra_properties)
-            if linked:
-                for module in app_copy.modules:
-                    if isinstance(module, ReportModule):
-                        messages.error(request, _('This linked application uses mobile UCRs which '
-                                                  'are currently not supported. For this application to '
-                                                  'function correctly, you will need to remove those modules.'))
-                        break
-            return back_to_main(request, app_copy.domain, app_id=app_copy._id)
+
+                linked_app = LinkedApplication(
+                    name=data['name'],
+                    domain=link_domain,
+                    master=app_id,
+                    master_domain=master_domain,
+                )
+                linked_app.save()
+                return pull_master_app(request, link_domain, linked_app.get_id)
+            else:
+                extra_properties = {'name': data['name']}
+                app_copy = import_app_util(app_id_or_source, link_domain, extra_properties)
+                if linked:
```

Comment:
i could be misreading this code but it looks like this clause is impossible since this is in the else clause (line 375) of the `if linked` in line 355, whcih would also mean we never verify a lack of UCRs (unless that part was on purpose)

Reply:
7ee88be